src/coredata.cs

```diff
@@ -1392,6 +1452,151 @@ interface NSMigrationManager {
 		bool UsesStoreSpecificMigrationManager { get; set; }
 	}
 
+	[Abstract]
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface NSPersistentHistoryChange : NSCopying
+	{
+		[Export ("changeID")]
+		long ChangeId { get; }
+
+		[Export ("changedObjectID", ArgumentSemantic.Copy)]
+		NSManagedObjectID ChangedObjectId { get; }
+
+		[Export ("changeType")]
+		NSPersistentHistoryChangeType ChangeType { get; }
+
+		[NullAllowed, Export ("tombstone", ArgumentSemantic.Copy)]
+		NSDictionary Tombstone { get; }
+
+		[NullAllowed, Export ("transaction", ArgumentSemantic.Strong)]
+		NSPersistentHistoryTransaction Transaction { get; }
+
+		[NullAllowed, Export ("updatedProperties", ArgumentSemantic.Copy)]
+		NSSet<NSPropertyDescription> UpdatedProperties { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface NSPersistentHistoryToken : NSCopying //, NSSecureCoding TODO: The class does state that it supports the NSSecureCoding YET SupportsSecureCoding returns false, radar 32761925
+	{
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSPersistentStoreRequest))]
+	interface NSPersistentHistoryChangeRequest
```

Comment:
Unless it make sense (base types) I think we should disable the default .ctor
Easier to add later than to remove a broken API